vault/identity_store_oidc_provider.go

```diff
@@ -50,15 +50,17 @@ type provider struct {
 }
 
 type providerDiscovery struct {
-	AuthorizationEndpoint string   `json:"authorization_endpoint"`
-	IDTokenAlgs           []string `json:"id_token_signing_alg_values_supported"`
 	Issuer                string   `json:"issuer"`
 	Keys                  string   `json:"jwks_uri"`
+	AuthorizationEndpoint string   `json:"authorization_endpoint"`
+	TokenEndpoint         string   `json:"token_endpoint"`
+	UserinfoEndpoint      string   `json:"userinfo_endpoint"`
+	IDTokenAlgs           []string `json:"id_token_signing_alg_values_supported"`
 	ResponseTypes         []string `json:"response_types_supported"`
 	Scopes                []string `json:"scopes_supported"`
 	Subjects              []string `json:"subject_types_supported"`
-	TokenEndpoint         string   `json:"token_endpoint"`
-	UserinfoEndpoint      string   `json:"userinfo_endpoint"`
+	GrantTypes            []string `json:"grant_types_supported"`
+	AuthMethods           []string `json:"token_endpoint_auth_methods_supported"`
```

Comment:
I commented here: https://github.com/hashicorp/vault/pull/12538#discussion_r715678579

But I think we should also include `request_uri_parameter_supported=false` in the Discovery endpoint response. The spec states that the default is `true` if it is not included. https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata

Reply:
Great catch! I agree. Adding this now.

Reply:
Added in 7c85382.